docs/e2e-test-guide.md

```diff
@@ -0,0 +1,164 @@
+# Argo CD Operator E2E Test Guide
```

Comment:
There's a short section in the development guide that talks about e2e testing.  That should be updated, maybe to point to this document?

https://github.com/argoproj-labs/argocd-operator/blob/85425ac53dfac5cf2f6c88b0bd481307cc053563/docs/contribute/development.md?plain=1#L80

Reply:
Ack